app/ui/src/app/customizations/api-connector/api-connector-info/api-connector-info.component.ts

```diff
@@ -52,63 +60,77 @@ export class ApiConnectorInfoComponent implements OnInit {
       this.apiConnectorDataForm.get('description').setValue(description);
       this.apiConnectorDataForm.get('host').setValue(configuredProperties.host || properties.host.defaultValue);
       this.apiConnectorDataForm.get('basePath').setValue(configuredProperties.basePath || properties.basePath.defaultValue);
-      this.isDirty = true;
     }
-  }
 
-  onChange(event): void {
-    if (event.target && event.target.files) {
-      const fileList = event.target.files as FileList;
-      if (fileList.length > 0) {
-        this.iconFile = fileList[0];
-      }
-    }
-    this.isDirty = true;
-    // If component is not in "Create" mode (eg. detail page), updating
-    // any input field will automatically fire up the submit handler
     if (!this.createMode) {
-      this.onSubmit();
+      this.apiConnectorDataForm.disable();
     }
   }
 
-  onSubmit(): void {
-    if (this.apiConnectorDataForm.valid && this.isDirty) {
-      const { name, description, host, basePath } = this.apiConnectorDataForm.value;
-      const apiConnectorData = {
-        ...this.apiConnectorData,
-        configuredProperties: {
-          ...this.apiConnectorData.configuredProperties,
-          host,
-          basePath,
-        },
-        name,
-        description,
-        iconFile: this.iconFile
-      } as CustomConnectorRequest;
-
-      this.update.emit(apiConnectorData);
-    }
-    this.editControlKey = null;
+  editForm() {
+    this.enableEdit();
   }
 
-  onEditEnable(event: Event, key: string): void {
-    event.preventDefault();
-    event.stopPropagation();
-    this.editControlKey = key;
+  enableEdit() {
+    this.editMode = true;
+    this.apiConnectorDataForm.enable();
```

Comment:
Can probably grab the values from `apiConnectorDataForm` here like:

`this.oldValues = { ...apiConnectorDataForm.value };`

Then when the user clicks 'Cancel':

`this.apiConnectorDataForm.value = this.oldValues`

which I think *should* reset the values displayed in the form.

Reply:
looks like `value` is a read only property, so I can't overwrite it, but I can use `setValue()`.